aws/aws-parallelcluster-cookbook
# [develop] Adapt base recipes to configure LoginNodes

### Description of changes
Each one of the following has a dedicated commit to make it easier to understand the PR

* Small preliminary fixes
    * Define helper method to identify LoginNodes 
    * Remove _netdev check from raid volumes test
    * Add LoginNode role in CloudWatch agent configuration 

* Cluster User
    * Configure cluster user in LoginNodes
    * Test cluster user configuration in LoginNodes 

* EBS volumes
    * Mount EBS volumes in LoginNodes
    * Test ebs volume configuration in LoginNodes 

* raid volumes
    * Mount raid volumes in LoginNodes
    * Test raid volume configuration in LoginNodes 

* dedicated shared folder for login nodes
    * Create shared_login_nodes directory 
    * Export shared_login_nodes directory 
    * Test /opt/intel folder configuration in LoginNodes 

* mounting of home and shared folders
    * Mount home and shared_login_nodes directories in login nodes 
    * Extract mount /home as subrecipe 
    * Test shared volumes configuration in LoginNodes 

* Slurm related configs
    * Setup munge and mount /opt/slurm in login nodes 
    * Validate that /opt/slurm is mounted in Compute and Login nodes 

Note the above test should be extended to validate also the munge configuration recipe. I did an attempt to create a mock environment for the test but the munge service still wasn't starting. Since the recipe is working (as my manual tests showed) I considered out-of-scope for this PR spending further time on this mock environment.

* Fixes in the logic used to retrieve the cluster config
    * Define current and previous config path for LoginNodes 
    * Ensure cluster config is retrieved only by the HeadNode 
    * LoadClusterConfig take the config path as parameter 

### Tests
* Most of the changes were successfully tested with unit tests
* I also manually launched a cluster with these changes in the recipes plus others required in the parallelcluster package and validated that I was able to log into a login node and have basic interactions with slurm (querying the cluster config, submitting a job, querying job status and job result)

### References
* [Changes in parallelcluster package](https://github.com/aws/aws-parallelcluster/pull/5446).

### Checklist
